cli/chaosli/cmd/context.go

```diff
@@ -0,0 +1,393 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/briandowns/spinner"
+	goyaml "github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	v1 "k8s.io/api/core/v1"
+	"log"
+	"math"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const MAXTARGETSHOW = 10
+const POD = "pod"
+
+func contextTargetsSize(disruption v1beta1.Disruption) ([]string, error) {
+	spec := disruption.Spec
+	labels := spec.Selector.String()
+	level := spec.Level
+	// If node level, this \n will give us empty namespaces for node which is intended because nodes
+	// do not require namespaces to be described
+	podNamespaces := disruption.ObjectMeta.Namespace
+	rowNames := 1
+
+	if level == "node" {
+		rowNames = 1
+	}
+
+	fmt.Println("Let's look at your targets...")
+
+	s := spinner.New(spinner.CharSets[38], 100*time.Millisecond)
+	s.Start()
+	cmd := fmt.Sprintf("kubectl get %s -n %s -l %s | wc -l", level, podNamespaces, labels)
+	sizeString, err := exec.Command("bash", "-c", cmd).Output()
+
+	if err != nil {
+		return nil, fmt.Errorf("Could not count the number of pods correlating to target selector: %v", err)
+	}
+	size, err := strconv.Atoi(strings.Trim(string(sizeString), "\n"))
+
+	if err != nil {
+		return nil, fmt.Errorf("Could not convert string to integer in context target size: %v", err)
+	}
+
+	// Remove header NAME from consideration
+	size = size - 1
+	// If the size is 0, first check if changing the level will do anything, otherwise
+	// mention to the user that the labels they are using won't target anything
+
+	if size <= 0 {
+		errorString := fmt.Sprintf("\nThe label selectors chosen (%s) result in 0 targets, meaning this disruption would do nothing given the namespace/cluster/label combination.", labels)
+		newLevel := ""
+
+		if level == POD {
+			newLevel = "node"
+		} else {
+			newLevel = POD
+		}
+		cmd := fmt.Sprintf("kubectl get %s -n %s -l %s | wc -l", newLevel, podNamespaces, labels)
+		sizeString, err := exec.Command("bash", "-c", cmd).Output()
+
+		if err != nil {
+			return nil, fmt.Errorf("Could not count the number of pods correlating to target selector: %v", err)
+		}
+		size, err = strconv.Atoi(strings.Trim(string(sizeString), "\n"))
+
+		if err != nil {
+			return nil, fmt.Errorf("Could not convert string to integer in context target size: %v", err)
+		}
+
+		if size > 0 {
+			errorString = fmt.Sprintf("\nWe noticed that your target size is 0 for level %s given your label selectors. We checked to see if the %s level would give you results and we found %d %ss. Is this the level you wanted to use?", level, newLevel, size, newLevel)
+		}
+		fmt.Println(errorString)
+		return nil, fmt.Errorf("bad sized targets")
```

Comment:
this error string could be improved

Reply:
I have the actual error string printed before the error fires. I didnt want to put errorString in the Errorf because it becomes a bit unclear with all the error string nesting. Do you think "bad sized targets" should be changed considering errorString has more informative information?

Reply:
A user running the context cmd is going to see
```
$printed_errorstring
Could not grab context regarding size and names of pods: bad sized targets
```
Why print errorstring separately rather than just returning error string?

Reply:
Thats fair, ok I'll just return the errorString.